sqsworkers/base.py

```diff
@@ -0,0 +1,276 @@
+            ).get_queue_by_name(QueueName=queue_name)
+            if sqs_resource is None
+            else sqs_resource
+        )
+
+        self.name = name or "crew-{}-{}-{}".format(
+            os.getpid(), (queue_name or next(self._count)), time.time()
+        )
+
+        self.logger = logging.LoggerAdapter(
+            (logging.getLogger() if logger is None else logger),
+            extra={"extra": {"crew.name": self.name}},
+        )
+
+        self.logger.setLevel(log_level)
+
+        statsd = StatsDBase(logger=logger) if statsd is None else statsd
+
+        assert issubclass(
+            statsd.__class__, interfaces.StatsDInterface
+        ), f"{statsd.__class__} does not conform to {interfaces.StatsDInterface.__name__}"
+
+        self.statsd = statsd
+        self.sentry = sentry
+        self.max_number_of_messages = max_number_of_messages
+        self.wait_time = wait_time
+        self.polling_interval = polling_interval
+
+    def start(self):
+
+        while True:
+
+            messages = self.queue.receive_messages(
+                AttributeNames=["All"],
+                MessageAttributeNames=["All"],
+                MaxNumberOfMessages=self.max_number_of_messages,
+                WaitTimeSeconds=self.wait_time,
+            )
+
+            if messages:
+
+                metadatas = [MessageMetadata(m) for m in messages]
+
+                self.logger.info(
+                    f"processing the following {len(messages)} messages: {metadatas}"
+                )
+
+                for message in messages:
+                    metadata = MessageMetadata(message)
+
+                    self.logger.info(f"processing message: {metadata}")
+
+                    task: futures.Future = self._executor.submit(
+                        self.message_processor(message).start
+                    )
+
+                    self.logger.info(f"processing task: {task}")
+
+                    task.add_done_callback(
+                        partial(
+                            self._task_complete,
+                            message=message,
+                            metadata=metadata,
+                        )
+                    )
+
+                    self.statsd.increment("process.record.start", 1, tags=[])
+
+            time.sleep(self.polling_interval)
+
+    def _task_complete(
+        self,
+        f: futures.Future,
+        message: Any,
+        metadata: Optional[MessageMetadata] = None,
+    ):
+        """
+        Clean up after the task and do any necessary logging.
+        """
+
+        metadata = MessageMetadata(message) if metadata is None else metadata
+
+        exception: Optional[Exception] = f.exception()
+
+        if exception is not None:
+
+            self.logger.error(
+                "failed processing {message} with the following exception: {exception}".format(
+                    exception=repr(exception), message=metadata
+                ),
+                exc_info=exception,
```

Comment:
Will this work with non-micros services and applications?

Reply:
yep

Reply:
👍